bip-tx-terminology.mediawiki

```diff
@@ -0,0 +1,291 @@
+
+; Script Tree
+: [Concept] The Merkle tree composed from the collection of Leaf Scripts committed to by a P2TR output
+
+; scriptPubKey
+: See _Output Script_
+
+; scriptSig
+: See _Input Script_
+
+; Sequence
+: [Component] The sequence number appearing in each transaction input.  Serialized as the four-byte <code>nSequence</code> integer field.  If any input’s sequence number is less than the maximum allowed value, a transaction is non-final.
+
+; Signature
+: [Component] A script argument that proves a key owner’s consent to a transaction. Composed of an r-value, an s-value, and a sighash flag. ECDSA and Schnorr signatures use different serialization schemes for signatures. The length indicator preceding the signature is a serialization artifact that is not part of the signature. If the default `SIGHASH_ALL` is used the sighash flag may be omitted for Schnorr signatures.
+
+; Sighash Flag
+: [Component] Part of the Signature. A one-byte field that indicates which parts of the transaction a signature commits to.
+
+; Taproot Branch
+: [Component] Part of a Control Block that specifies the hashing partners necessary to prove the existence of a Leaf Script in a Script Tree
+
+; Taproot Internal Key
+: [Component] aka “Inner Key”. The untweaked public key before it is tweaked with the Merkle root of the Script Tree to create the Taproot Output Key. Appears as part of a Control Block for a script path P2TR input.
+
+; Taproot Output Key
+: [Component] The public key present in the Witness Program of a P2TR output script composed by tweaking the Inner Key with the Script Tree's Merkle root
+
+; Tapscript
+: [Concept] The variant of the Script language used in P2TR Leaf Scripts (see BIP342)
+
+; Transaction Header
+: [Concept] Collective term for the serialization artifacts and transaction fields that appear only once in the transaction serialization regardless of lengths of the Input and Output Lists: Transaction Version, Marker (segwit only), Flag (segwit only), Input Counter, Output Counter, Lock Time
+
+; Transaction Input
+: [Component] aka Input. An element of the input list of the transaction. Must include an outpoint, an input script, and a sequence. The input script can have various content, corresponding to the UTXO that is being spent. The input script has length 0 for native segwit inputs. In segwit transactions, every input must be accompanied by a Witness Stack. The Witness Stack is not part of the Transaction Input.
+
+; Transaction Input List
+: [Concept] The enumeration of all Transaction Inputs of a transaction
+
+; Transaction Output
+: [Component] Creates a UTXO by specifying an amount and an output script
+
+; Transaction Output List
+: [Concept] The enumeration of all Transaction Outputs of a transaction
+
+; txid
+: [Component] Part of an Outpoint that identifies the transaction that created a UTXO. Result of hashing a transaction without its witness structure.
+
```

Comment:
```suggestion
; UTXO
; [Component] A UTXO (Unspent Transaction Output) refers to unspent outputs of a previous transaction.
```

Reply:
I notice we use UTXO in various parts of this doc but haven't defined it explicitly.